resolwe_bio/expression_filters/relation.py

```diff
@@ -0,0 +1,67 @@
+""".. Ignore pydocstyle D400.
+
+=======================
+Relations Template Tags
+=======================
+
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from resolwe.flow.expression_engines.jinja.filters import id_
+from resolwe.flow.models.entity import RelationPartition
+
+from resolwe_bio.models import Sample
+
+
+def replicate_groups(data):
+    """Get replicate groups."""
+    if not isinstance(data, list):
+        raise ValueError('List of data IDs must be given')
+
+    data_ids = [id_(d) for d in data]
+
+    if len(data_ids) != len(set(data_ids)):
+        raise ValueError('Repeated data objects not allowed')
+
+    samples = Sample.objects.filter(data__id__in=data_ids)
+
+    if len(samples) != len(data_ids):
+        raise ValueError('Can not get replicates of data without sample')
+
+    partitions = RelationPartition.objects.filter(relation__category='Replicate',
```

Comment:
This format might be better: https://github.com/genialis/resolwe-bio/pull/657/files#diff-fbfd490016a4d3a3217974dcdf871ff9R98